File: src/lib/tauri-api/workspace-cleanup.ts
import type { CleanupWorkspaceInput, CleanupWorkspaceResult } from '../../types';
import { invokeCommand } from './client';

export function cleanupWorkspace(input: CleanupWorkspaceInput): Promise<CleanupWorkspaceResult> {
  return invokeCommand<CleanupWorkspaceResult>('cleanup_workspace', { input });
}

File: frontend/src-tauri/src/commands/workspace_cleanup.rs
use tauri::State;

use crate::models::{CleanupWorkspaceInput, CleanupWorkspaceResult};
use crate::services::workspace_cleanup_service;
use crate::state::AppState;

#[tauri::command]
pub fn cleanup_workspace(
    state: State<'_, AppState>,
    input: CleanupWorkspaceInput,
) -> Result<CleanupWorkspaceResult, String> {
    workspace_cleanup_service::cleanup_workspace(&state, input)
}
